wdl-engine/src/inputs.rs

```diff
@@ -27,6 +28,90 @@ use crate::Value;
 /// A type alias to a JSON map (object).
 type JsonMap = serde_json::Map<String, JsonValue>;
 
+/// Trait for parsing different input formats (JSON, YAML, etc.)
+///
+/// This trait defines an interface for parsing different file formats into a common
+/// representation (JsonMap) that can be used by the inputs system.
+pub(crate) trait InputFormat {
+    /// Parse a file into a JsonMap that can be used by the inputs system
+    /// 
+    /// # Parameters
+    /// - `path`: Path to the input file
+    ///
+    /// # Returns
+    /// - A JsonMap containing the parsed input data
+    ///
+    /// # Errors
+    /// - If the file cannot be opened or read
+    /// - If the file's content cannot be parsed
+    /// - If the parsed content is not a valid object/mapping
+    fn parse_file(path: &Path) -> Result<JsonMap>;
+}
+
+/// JSON input format implementation
+///
+/// Parses JSON files into the common JsonMap format.
+pub(crate) struct JsonInputFormat;
+
+impl InputFormat for JsonInputFormat {
+    fn parse_file(path: &Path) -> Result<JsonMap> {
+        let file = File::open(path)
+            .with_context(|| format!("failed to open input file `{path}`", path = path.display()))?;
+        
+        // Parse the JSON (should be an object)
+        let reader = BufReader::new(file);
+        let object = mem::take(
+            serde_json::from_reader::<_, JsonValue>(reader)
+                .with_context(|| {
+                    format!("failed to parse input file `{path}`", path = path.display())
+                })?
+                .as_object_mut()
+                .with_context(|| {
+                    format!(
+                        "expected input file `{path}` to contain a JSON object",
+                        path = path.display()
+                    )
+                })?,
+        );
+        
+        Ok(object)
+    }
+}
+
+/// YAML input format implementation
+///
+/// Parses YAML files into the common JsonMap format.
+pub(crate) struct YamlInputFormat;
+
+impl InputFormat for YamlInputFormat {
+    fn parse_file(path: &Path) -> Result<JsonMap> {
+        let file = File::open(path)
+            .with_context(|| format!("failed to open input file `{path}`", path = path.display()))?;
+        
+        // Parse the YAML
+        let reader = BufReader::new(file);
+        let yaml: YamlValue = serde_yaml::from_reader(reader)
+            .with_context(|| format!("failed to parse input file `{path}`", path = path.display()))?;
+        
+        // Convert YAML to JSON format
+        let mut json = serde_json::to_value(yaml)
+            .with_context(|| format!("failed to convert YAML to JSON for processing"))?;
```

Comment:
There's two issues here: a use of `format!` with no interpolations (i.e. `{}`) is unnecessary and `with_context` is unnecessary if you want to use an unformatted string; use `context` instead:

```suggestion
            .context("failed to convert YAML to JSON for processing")?;
```